lib/uklibid/Config.uk

```diff
@@ -1,10 +1,113 @@
-config LIBUKLIBID
+menuconfig LIBUKLIBID
 	bool "uklibid: Library identifier"
-	default n
+	default y
 	# FIXME: We actually depend on <uk/assert.h> but lib/ukdebug also
 	#        depends on us. If we specify the dependency here, it creates a
 	#        circular dependency that KConfig doesn't like. However, we
 	#        can currently assume that lib/ukdebug is always part of a
 	#        build, so we comment out the dependency here. Of course, this
 	#        is not accurate.
 	#select LIBUKDEBUG
+
+if LIBUKLIBID
+config LIBUKLIBID_INFO
+	bool "Embedded library metadata"
+	default y
+
+	if LIBUKLIBID_INFO
+	menu "Global metadata"
+		config LIBUKLIBID_INFO_UKFULLVERSION
+			bool "Full unikraft version"
+			default y
+			help
+				If enabled, the full Unikraft version including
+				Git SHA is embedded. Otherwise, only the short
+				Unikraft version is stored.
+
+		config LIBUKLIBID_INFO_UKCODENAME
+			bool "Unikraft codename"
+			help
+				If enabled, the Unikraft release codename is
+				stored as part of the global metadata
+				information.
+
+		config LIBUKLIBID_INFO_UKCONFIGGZ
+			bool "Embed gzipped configuration (.config)"
+			help
+				If enabled, the current configuration `.config`
+				of the build is gzipped and embedded to the
+				global metadata infomration space.
+	endmenu
+
+	menu "Per library metadata"
+		config LIBUKLIBID_INFO_LIB_UKVERSION
+			bool "Unikraft version"
+			help
+				If enabled, the Unikraft version is included
+				to each library.
+
+		config LIBUKLIBID_INFO_LIB_UKFULLVERSION
+			bool "Full version"
+			depends on LIBUKLIBID_INFO_LIB_UKVERSION
+			help
+				Instead of the short Unikraft version, the full
+				Unikraft version including Git SHA is embedded
+				to each library.
+
+		config LIBUKLIBID_INFO_LIB_UKCODENAME
+			bool "Unikraft codename"
+			help
+				If enabled, the Unikraft release codename is
+				stored as part of each library metadata
+				information.
+	endmenu
+
+	config LIBUKLIBID_INFO_COMPILER
```

Comment:
These options are neither global nor per library, both here and in `libinfo.S`. What is the rationale for that? Moreover, with the possibility of external libraries provided as prebuilts, it is possible that a library provided as a binary all the compiler info may be different than the rest of the build. Therefore, should these be considered as per-library data too?

Reply:
This info was stored in both places: per-library and globally. HOwever, to reduce confsion, I am going to change it and add the configuration items under both, the global configuration and per-library.